packages/amqp/lib/AbstractAmqpPublisher.ts

```diff
@@ -0,0 +1,94 @@
+import type { Either } from '@lokalise/node-core'
+import type {
+  BarrierResult,
+  MessageInvalidFormatError,
+  MessageValidationError,
+  QueuePublisherOptions,
+  SyncPublisher,
+} from '@message-queue-toolkit/core'
+import { MessageSchemaContainer, objectToBuffer } from '@message-queue-toolkit/core'
+import type { ZodSchema } from 'zod'
+
+import type { AMQPLocator, AMQPCreationConfig, AMQPDependencies } from './AbstractAmqpService'
+import { AbstractAmqpService } from './AbstractAmqpService'
+
+export type AMQPPublisherOptions<MessagePayloadType extends object> = QueuePublisherOptions<
+  AMQPCreationConfig,
+  AMQPLocator,
+  MessagePayloadType
+>
+
+export abstract class AbstractAmqpPublisher<MessagePayloadType extends object>
```

Comment:
This is the old `AbstractAmqpBasePublisher` combined with `AbstractAmqpPublisherMultiSchema`